### DISCUSSION TITLE
Understanding more FaceRecognation

### DISCUSSION THREAD

**Discussion by ewok2** [matiasdelellis/facerecognition#742] [Q&A] [answered]
Hello
I have try to read all the Wiki to have a better understanding of how FaceRecognation works, but still have some questions:

1- Am I right if I say that the PDlib model has been already train but will not take into account for the new photo comming the manual reassignation perform ? 

=> For exemple to be more clear
If FaceRecognation detect 10 face of person1 and 10 faces of person2, but 2 faces tag as person1 are in fact person2.
I will set manualy this 2 faces as personn2
 >> if I put a knew picture with almost the same face as the 2 with a bad assignation, will this new face be tag as person1 (and that mean it does not take into account the previous categorisation) or it will be assigned to person2 (It will be ideal but I have doubt...)

2- It is not clear for me what append with X photos already process if I change the parameter of the model (to get more face detected for exemple). In the "Functional spec" chapter I am not sure if what is writen is wishes for the futur or already reality ? (This is probably due to my bad english enderstanding...)
What actions will have an impact of the manual tagging and personn groupping?

Manys thanks

**Comment by matiasdelellis** [comment, marked as answer]
> 1- Am I right if I say that the PDlib model has been already train but will not take into account for the new photo comming the manual reassignation perform ?

Indeed in a pre-trained model, which unfortunately we do not have the ability to retrain with our data.

> If FaceRecognation detect 10 face of person1 and 10 faces of person2, but 2 faces tag as person1 are in fact person2.
I will set manualy this 2 faces as personn2

If you named person 2, and then removed the faces that were wrong using the "This is not person 2" option these two faces are fixed to person 1, but this correction is not taken into account to improve the results.

> In the "Functional spec" chapter I am not sure if what is writen is wishes for the futur or already reality ? (This is probably due to my bad english enderstanding...)

The Functional spec represents the current use quite well, but is probably out of date in some ways. Don't know. I'd have to review it.

> What actions will have an impact of the manual tagging and personn groupping?

We are currently focused on automatic clustering, which has some drawbacks.
* It is generally stable, but due to certain heuristics the result may change and some named people may disappear. We have to name them again.
* Regarding the actions available to the user, the name is respected, and as I mentioned, if the option "This is not such a person" is used, it face is separated from the wrong cluster and does does not try to recluster with other faces..

**Comment by ewok2** [comment]
Thanks for all this answer
As I am speeking to one of the father of FaceRecognation ... wonderful app!!!
Did you intend to produce a version compatible with NC28 or 29 ?
